common/test/acceptance/tests/lms/test_account_settings.py

```diff
@@ -233,20 +247,13 @@ def test_full_name_field(self):
             [u'another name', self.username],
         )
 
-        self.verify_settings_changed_events(
-            self.username, self.user_id,
+        actual_events = self.wait_for_events(event_filter=self.settings_changed_event_filter, number_of_matches=2)
+        self.assert_event_sequences_match(
```

Comment:
I now see that this test follows the wait then assert pattern I mentioned above. I love how this preserves the readability of my original code, while providing a much more powerful framework. I suggest reworking ``test_page_view_event`` above to have the same two step logic, where the first step captures all similar events, and the second step asserts that the captured events are as expected.